### PR TITLE
7421 change of name certificate - if the company changes its legal name

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
*Issue #:* /bcgov/entity#7421

*Description of changes:*
  - 7421 change of name certificate - if the company changes its legal name
  - 6766 bug - pay api receipt

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/services/document_meta.py [hunk 1]
```diff
@@ -286,7 +286,10 @@ def get_alteration_reports(self, filing: dict):  # pylint: disable=no-self-use
                     DocumentMetaService.ReportType.ALTERATION_NOTICE.value
                 )
             )
-            if 'nameRequest' in filing['filing']['alteration']:
+            name_request = filing.get('filing', None).get('alteration', None).get('nameRequest', None)
+            business = filing.get('filing', None).get('business', None)
+            if name_request and 'legalName' in name_request and \
+                    name_request['legalName'] != business.get('legalName', None):
```
Comment (andrepestana-aot) [inline legal-api/src/legal_api/services/document_meta.py:292]: Did you check the change of name to a numbered company? That should generate a certificate as well.

**Comment by andrepestana-aot** [comment]
Can you please add any tests to check the change of name certificate?

**Review by andrepestana-aot — CHANGES_REQUESTED** [review]
Added some comments

File: legal-api/src/legal_api/services/document_meta.py (hunk 1 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/services/document_meta.py:292]: Yes I did check the existing filing json. This should work

**Review by kialj876 — APPROVED** [review]
Looks good, I just have a couple comments things to consider

File: queue_services/entity-emailer/src/entity_emailer/email_processors/filing_notification.py [hunk 2]
```diff
@@ -287,6 +294,9 @@ def get_additional_info(filing: Filing) -> dict:
         if original_filing_type == 'incorporationApplication':
             additional_info['nameChange'] = NameXService.has_correction_changed_name(filing.filing_json)
     elif filing.filing_type == 'alteration':
-        additional_info['nameChange'] = 'nameRequest' in filing.filing_json['filing']['alteration']
+        name_request = filing.filing_json.get('filing', None).get('alteration', None).get('nameRequest', None)
```
Comment (kialj876) [inline queue_services/entity-emailer/src/entity_emailer/email_processors/filing_notification.py:297]: same thing here

**Review by andrepestana-aot — APPROVED** [review]
LGTM

**Pull request merged**